internal/addon/addon.go

```diff
@@ -30,3 +39,83 @@ func GetObjectKey(configRef []addonapiv1alpha1.ConfigReference, group, resource
 	}
 	return key
 }
+
+func AgentHealthProber() *agent.HealthProber {
+	return &agent.HealthProber{
+		Type: agent.HealthProberTypeDeploymentAvailability,
+		WorkProber: &agent.WorkHealthProber{
+			ProbeFields: []agent.ProbeField{
+				{
+					ResourceIdentifier: workapiv1.ResourceIdentifier{
+						Group:     OtelcolGroup,
+						Resource:  OtelcolResource,
+						Name:      OtelcolName,
+						Namespace: OtelcolNS,
```

Comment:
Do we need explicitly to set both Name & Namespace?

Reply:
Apparently we don't.. I just checked the api and the namespace is optional..